### ISSUE TITLE
[FG:InPlacePodVerticalScaling] Pod Resize - Investigate long delay in updating apiPodStatus.Resources

### ISSUE THREAD

**Issue by vinaykul** [kubernetes/kubernetes#112264]
### What would you like to be added?

This is really a bug. 

Running pod resize E2E tests full scope in GKE 1-master 1-worker setup takes significantly longer (869s vs 2988s  to run all 34 E2E tests) Initial investigation shows the the issue is NOT in containerd response to ContainerStatus() CRI API that is called within 50 millisec of UpdateContainerResources(). The long delay is in updating apiPodStatus.

### Why is this needed?

There is no good reason I can think of why it should take several 10s of seconds to update the apiPodStatus.Resources.

**Comment by Jeffwan** [comment]
@vinaykul I figure out the problem in our internal version the major reason is the cgroup update and reconcile are separate loops. The first reconcile loop after the update can not retrieve the up-to-date status. This leads a problem that it has to rely on kubelet `—sync-frequency` (default 60s) to sync the pod and then update the status. I can help commit the improvement

**Comment by Jeffwan** [comment]
@vinaykul I try to understand the source of the performance data.  `869s vs 2988s` this is the before/after from prow job   pull-kubernetes-e2e-gce-ubuntu-containerd?  test `Kubernetes e2e suite: [It] [sig-node] Pod InPlace Resize Container [Feature:InPlacePodVerticalScaling]`

**Comment by vinaykul** [comment]
The 2988s run time comes for the additional time it takes across all the 30+ different E2E test cases when full E2E testing is enabled. (Disable isFeatureGatePostAlpha() check which runs the test in full-E2E mode where it checks for ContainerStatus.Resources == ContainerSpec.Resources - this requires containerd support. I tested it by replacing the containerd binaries with those I build from latest containerd master)

**Comment by vinaykul** [comment]
hi @Jeffwan , just curious, how is it looking with this issue?

**Comment by SergeyKanzhelev** [comment]
/remove-lifecycle rotten

Is it still a valid issue?

**Comment by vinaykul** [comment]
Yes, it is still an issue. @Jeffwan Do you have cycles for this?

**Comment by SergeyKanzhelev** [comment]
is it something that will be blocking for 1.27 release?

**Comment by sftim** [comment]
/retitle [FG:InPlacePodVerticalScaling] Pod Resize - long delay in updating `apiPodStatus.Resources`

It's a bug, right?
/kind bug

**Comment by SergeyKanzhelev** [comment]
/priority important-longterm
/triage accepted
@Jeffwan are you working on this or want to be unassigned?

**Comment by jbartosik** [comment]
This bug makes it slightly harder to use in-place updates in VPA (https://github.com/kubernetes/autoscaler/issues/4016).

**Comment by Jeffwan** [comment]
sorry I missed a few messages. We are preparing a PR for this issue and it will be submitted next week. Currently, we are working on master branch testing

**Comment by SergeyKanzhelev** [comment]
/triage accepted

Still an issue for the alpha feature

**Comment by tallclair** [comment]
Marking this as a duplicate of https://github.com/kubernetes/kubernetes/issues/123940

/close
